com.unity.render-pipelines.core/Editor/Lighting/LightAnchorEditor.cs

```diff
@@ -0,0 +1,662 @@
+
+                Event.current.Use();
+                GUI.changed = true;
+            }
+            else if (enabled)
+            {
+                var srcPos = new Vector2(
+                    Mathf.Cos((angle + offset) * Mathf.Deg2Rad),
+                    Mathf.Sin((angle + offset) * Mathf.Deg2Rad)) * state.radius + state.position;
+
+                var dstPos = Slider2D(id, srcPos, 5f, Handles.CircleHandleCap);
+                dstPos -= state.position;
+                dstPos.Normalize();
+
+                newAngle = LightAnchor.NormalizeAngleDegree(Mathf.Atan2(dstPos.y, dstPos.x) * Mathf.Rad2Deg - offset);
+                newAngle = Mathf.Round(newAngle * 100.0f) / 100.0f;
+            }
+            else
+            {
+                newAngle = 0;
+            }
+
+            if (Event.current.type == EventType.Repaint)
+            {
+                DrawAngleWidget(state.position, state.radius, newAngle, offset, sectionColor, enabled);
+            }
+
+            return newAngle;
+        }
+
+        static void DrawAngleWidget(Vector2 center, float radius, float angleDegrees, float offset, Color sectionColor, bool enabled)
+        {
+            Vector2 originPosition = center + new Vector2(
+                Mathf.Cos(offset * Mathf.Deg2Rad),
+                Mathf.Sin(offset * Mathf.Deg2Rad)) * radius;
+
+            Vector2 toOrigin = originPosition - center;
+
+            Vector2 handlePosition = center + new Vector2(
+                Mathf.Cos((angleDegrees + offset) * Mathf.Deg2Rad),
+                Mathf.Sin((angleDegrees + offset) * Mathf.Deg2Rad)) * radius;
+
+            Color backupColor = Handles.color;
+            Handles.color = LightAnchorStyles.DiskBackgroundAngleColor();
+            Handles.DrawSolidDisc(center, Vector3.forward, radius);
+            Handles.color = LightAnchorStyles.angleDiskBorderColor;
+            Handles.DrawWireDisc(center, Vector3.forward, radius);
+            Handles.color = sectionColor;
+            Handles.DrawSolidArc(center, Vector3.forward, Quaternion.AngleAxis(offset, Vector3.forward) * Vector3.right, angleDegrees, radius);
+            Handles.color = LightAnchorStyles.WireDiskAngleColor();
+            Handles.DrawLine(center + toOrigin * 0.75f, center + toOrigin * 0.9f);
+            Handles.DrawLine(center, handlePosition);
+            Handles.DrawSolidDisc(handlePosition, Vector3.forward, 5f);
+            Handles.color = backupColor;
+        }
+
+        static Rect SliceRectVertical(Rect r, float min, float max)
+        {
+            return Rect.MinMaxRect(
+                r.xMin, Mathf.Lerp(r.yMin, r.yMax, min),
+                r.xMax, Mathf.Lerp(r.yMin, r.yMax, max));
+        }
+
+        static Vector2 s_CurrentMousePosition;
+        static Vector2 s_DragStartScreenPosition;
+        static Vector2 s_DragScreenOffset;
+
+        static internal Vector2 Slider2D(int id, Vector2 position, float size, Handles.CapFunction drawCapFunction)
```

Comment:
I think we already have 2D sliders available : Handles.Slider2D(...). Can't it be used here?

Reply:
Tested but the behavior is not the one desired, can be renamed to avoir confusion